### PR TITLE
abolotski mini project

### PR DESCRIPTION
**Pull request by asherbolotski**
added my mini project of the y-axis on an XBox controller controlling the speed of a motor

### CODE REVIEW

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    configureButtonBindings();
+  }
+
+
+    /** This function is called periodically during operator control. */
+    public void teleopPeriodic() {
+      //if(mainController.getLeftY() * -1.0 >= Constants.XBoxStickTolerence * -1 && mainController.getLeftY() * -1.0 <= Constants.XBoxStickTolerence){
+        //new ScheduleCommand(new StopMotor(talonMotorSub));
+      //}
+      CommandScheduler.getInstance().schedule(new SetMotorSpeed(talonMotorSub, mainController.getLeftY() * -1.0));
```
Comment (mcadamswa) [inline camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java:45]: Can you explain what the magic 1.0 for at the end of this line?

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  /** The container for the robot. Contains subsystems, OI devices, and commands. */
+  public RobotContainer() {
+    // Configure the button bindings
+    configureButtonBindings();
+  }
+
+
+    /** This function is called periodically during operator control. */
+    public void teleopPeriodic() {
+      //if(mainController.getLeftY() * -1.0 >= Constants.XBoxStickTolerence * -1 && mainController.getLeftY() * -1.0 <= Constants.XBoxStickTolerence){
```
Comment (mcadamswa) [inline camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java:42]: The commented out code should either have a comment before it that explains why it is still needed / necessary and/or just delete it.

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    }
+  
+  /**
+   * Use this method to define your button->command mappings. Buttons can be created by
+   * instantiating a {@link GenericHID} or one of its subclasses ({@link
+   * edu.wpi.first.wpilibj.Joystick} or {@link XboxController}), and then passing it to a {@link
+   * edu.wpi.first.wpilibj2.command.button.JoystickButton}.
+   */
+  private void configureButtonBindings() {
+    //talonMotorSub.setDefaultCommand(new SetMotorSpeed(talonMotorSub, mainController.getLeftY() * -1.0));
```
Comment (mcadamswa) [inline camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java:55]: Seems like this line should be deleted.

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/ExampleCommand.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import frc.robot.subsystems.ExampleSubsystem;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+/** An example command that uses an example subsystem. */
+public class ExampleCommand extends CommandBase {
```
Comment (mcadamswa) [inline camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/ExampleCommand.java:11]: So do you need this file?

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.EpicTalonMotor;
+
+
```
Comment (mcadamswa) [inline camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java:7]: missing comments

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+public class SetMotorSpeed extends CommandBase{
+    private EpicTalonMotor epicTalonMotorSubsytem;
+    private double mySpeed = 0.0;
+    private boolean done = false;
+
+    public SetMotorSpeed(EpicTalonMotor epicTalonMotorSubsystem, double speed) {
```
Comment (mcadamswa) [inline camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java:13]: missing definition comment block

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        this.epicTalonMotorSubsytem = epicTalonMotorSubsystem;
+        this.mySpeed = speed;
+        addRequirements(epicTalonMotorSubsystem);}
+
+    public void initialize(){
```
Comment (mcadamswa) [inline camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java:18]: missing definition comment bloc

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/StopMotor.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+    public void execute(){
+        epicTalonMotorSubsytem.StopMotor();
+        done = true;}
+    
+    public void end(){
+    done = true;}
+
+    public boolean isFinished(){
+    return done;}
```
Comment (mcadamswa) [inline camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/StopMotor.java:25]: put the end curly brace on its own line

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/subsystems/ExampleSubsystem.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) FIRST and other WPILib contributors.
```
Comment (mcadamswa) [inline camp/abolotski/XBoxToTalon/src/main/java/frc/robot/subsystems/ExampleSubsystem.java:1]: I don't think we need this class file.

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java [hunk 10]
```diff
@@ -0,0 +1,63 @@
+// Copyright (c) FIRST and other WPILib contributors.
```
Comment (mcadamswa) [inline camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java:1]: What would be better is if we have a bit to describe the file here.  We have a block used in Yogi2 ... take a look, can you find a better block?

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+  private void configureButtonBindings() {
+    //talonMotorSub.setDefaultCommand(new SetMotorSpeed(talonMotorSub, mainController.getLeftY() * -1.0));
+  }
+
+  /**
+   * Use this to pass the autonomous command to the main {@link Robot} class.
+   *
+   * @return the command to run in autonomous
+   */
+}
```
Comment (mcadamswa) [inline camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java:63]: Where did the simulation stuff go?

File: camp/abolotski/XBoxToTalon/simgui-ds.json [hunk 12]
```diff
@@ -0,0 +1,97 @@
+{
```
Comment (josieammer) [inline camp/abolotski/XBoxToTalon/simgui-ds.json:1]: wondering if we need to checkin this file, or if it should be in .gitignore?

File: camp/abolotski/XBoxToTalon/simgui.json [hunk 13]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (josieammer) [inline camp/abolotski/XBoxToTalon/simgui.json:1]: wondering if we need to checkin this file, or if it should be in .gitignore?

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/Constants.java [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+ * constants. This class should not be used for any other purpose. All constants should be declared
+ * globally (i.e. public static). Do not put anything functional in this class.
+ *
+ * <p>It is advised to statically import this class (or one of its inner classes) wherever the
+ * constants are needed, to reduce verbosity.
+ */
+public final class Constants {
+    public static final int JoystickMotorCanId = 2;
+
+    public static final int PortMainController = 0;
```
Comment (josieammer) [inline camp/abolotski/XBoxToTalon/src/main/java/frc/robot/Constants.java:18]: could you give this a better name? I'm not sure what it's for from this name.

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+        System.out.println("in command initialization");
```
Comment (josieammer) [inline camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java:19]: remove debug print statements from final code.

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+        done = false;}
+
+    public void execute() {
+        System.out.println("in command execution" + mySpeed);
```
Comment (josieammer) [inline camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java:23]: remove debug print statements from final code.

**Review by josieammer — CHANGES_REQUESTED** [review]

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java (hunk 1 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java:45]: added comment

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java (hunk 2 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java:42]: deleted

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java (hunk 3 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java:55]: deleted

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java (hunk 5 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java:7]: added

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java (hunk 7 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java:18]: added

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/ExampleCommand.java (hunk 4 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/ExampleCommand.java:11]: no, deleted

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/subsystems/ExampleSubsystem.java (hunk 9 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/src/main/java/frc/robot/subsystems/ExampleSubsystem.java:1]: we don't, deleted

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java (hunk 6 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java:13]: added

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/StopMotor.java (hunk 8 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/StopMotor.java:25]: moved :/

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java (hunk 10 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java:1]: changed to the yogi2 block

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java (hunk 11 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java:63]: its not needed, WPIlib includes it as a default now (thats what my mom says) the sim still runs

File: camp/abolotski/XBoxToTalon/simgui-ds.json (hunk 12 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/simgui-ds.json:1]: added to .gitignore

File: camp/abolotski/XBoxToTalon/simgui.json (hunk 13 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/simgui.json:1]: added to .gitignore

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/Constants.java (hunk 14 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/src/main/java/frc/robot/Constants.java:18]: talked inperson

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java (hunk 15 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java:19]: removed

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java (hunk 16 above)
Comment (asherbolotski) [reply camp/abolotski/XBoxToTalon/src/main/java/frc/robot/commands/SetMotorSpeed.java:23]: removed

**Comment by asherbolotski** [comment]
@mcadamswa I've addressed all the comments and pushed the changes

File: camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java (hunk 11 above)
Comment (mmcadamsms) [reply camp/abolotski/XBoxToTalon/src/main/java/frc/robot/RobotContainer.java:66]: ok ... will try it once you merge ...

**Review by mmcadamsms — APPROVED** [review]

**Review by josieammer — APPROVED** [review]

**Pull request merged**